ddtrace/settings.py

```diff
@@ -102,8 +104,8 @@ def __init__(self, global_config, *args, **kwargs):
         :param kwargs:
         """
         super(IntegrationConfig, self).__init__(*args, **kwargs)
-
         self.global_config = global_config
+        self.hooks = Hooks()
 
     def __deepcopy__(self, memodict=None):
         new = IntegrationConfig(self.global_config, deepcopy(dict(self)))
```

Comment:
Should we handle the deep copy of hooks as well?

Reply:
oh yes, we need to, great catch!

Reply:
```suggestion
        new = IntegrationConfig(self.global_config, deepcopy(dict(self)))
        new.hooks = self.hooks
```